tests/IPv4NetworkSpecTest.php

```diff
@@ -54,6 +54,18 @@ public function testContainsWorksWithIPv6(): void
         self::assertFalse($network->contains(IPv6Address::parse('::ffff:808:808')));
     }
 
+    public function testContainsWorksWithNetworkInNetwork(): void
+    {
+        $network = IPv4Network::parse('127.0.0.1/16');
+
+        self::assertTrue($network->contains(IPv4Network::parse('127.0.0.1/16')));
+        self::assertTrue($network->contains(IPv4Network::parse('127.0.1.1/24')));
+        self::assertTrue($network->contains(IPv4Network::parse('127.0.0.4/23')));
+
+        self::assertFalse($network->contains(IPv4Network::parse('172.16.0.1/16')));
+        self::assertFalse($network->contains(IPv4Network::parse('127.0.0.1/8')));
```

Comment:
Could you add a test case for checking on IPv6 networks. An IPv4 network should never contain an IPv6 network

We can also track this in a separate issue if you want.

Reply:
Not creating another issue, doing it right the first time.